de.tudarmstadt.ukp.wikipedia.api/src/main/java/de/tudarmstadt/ukp/wikipedia/api/hibernate/CategoryDAO.java

```diff
@@ -19,110 +19,56 @@
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
-import org.hibernate.LockOptions;
-import org.hibernate.SessionFactory;
 
 import de.tudarmstadt.ukp.wikipedia.api.Wikipedia;
 
 /**
- * Data access object for class Page
- * @see de.tudarmstadt.ukp.wikipedia.api.Page
+ * Data access object for class {@link Category}
+ * 
+ * @see de.tudarmstadt.ukp.wikipedia.api.Category
  * @author Hibernate Tools
  */
-public class CategoryDAO {
+public class CategoryDAO extends GenericDAO<Category> {
 
 	private final Log logger = LogFactory.getLog(getClass());
 
-    private static final String CATEGORY_CLASS = "de.tudarmstadt.ukp.wikipedia.api.hibernate.Category";
-
-    private final SessionFactory sessionFactory;
-
-    private Wikipedia wiki;
-
     public CategoryDAO(Wikipedia pWiki) {
-        this.wiki = pWiki;
-        sessionFactory = getSessionFactory();
-    }
-
-    protected SessionFactory getSessionFactory() {
-        try {
-            return WikiHibernateUtil.getSessionFactory(wiki.getDatabaseConfiguration());
-        } catch (Exception e) {
-            logger.error("Could not locate SessionFactory in JNDI", e);
-            throw new IllegalStateException("Could not locate SessionFactory in JNDI");
-        }
+        super(pWiki, "de.tudarmstadt.ukp.wikipedia.api.hibernate.Category");
```

Comment:
How about using `Category.class.getName()` here? Or even just `Category.class`? (general comment for other DAO classes)

Reply:
I see the point and will try out both options.